packages/image_picker/image_picker_for_web/lib/image_picker_for_web.dart

```diff
@@ -310,6 +327,73 @@ class ImagePickerPlugin extends ImagePickerPlatform {
     _target.children.add(element);
     element.click();
   }
+
+  Future<String> _getCompressedUri(html.File file, int? imageQuality,
+      double? maxHeight, double? maxWidth) async {
+    //max width, max height, imageQuality are not supported for gif
+    if (file.type == "image/gif") {
+      return html.Url.createObjectUrl(file);
+    }
+    final Completer<String> _completer = Completer<String>();
+    final blobUrl = html.Url.createObjectUrl(file);
+    final image = html.ImageElement(src: blobUrl);
+    image.onLoad.listen((event) async {
+      html.Url.revokeObjectUrl(blobUrl);
+      final canvas = html.CanvasElement();
+      final size = _calculateSize(image, maxWidth ?? image.width!.toDouble(),
+          maxHeight ?? image.height!.toDouble());
+      canvas.width = size[0];
+      canvas.height = size[1];
+      final ctx = canvas.context2D;
+      if (maxHeight == null && maxWidth == null) {
+        ctx.drawImage(image, 0, 0);
+      } else {
+        ctx.drawImageScaled(image, 0, 0, canvas.width!, canvas.height!);
+      }
+      final blob = await canvas.toBlob(
+          file.type,
+          (imageQuality ?? 100) /
+              100.0); // Image quality only works for jpeg images
+      _completer.complete(html.Url.createObjectUrlFromBlob(blob));
+    });
+    image.onError.listen((event) {
+      //Return the original image if error comes
+      _completer.complete(html.Url.createObjectUrl(file));
+    });
+    return _completer.future;
+  }
+
+  List<int> _calculateSize(
+      html.ImageElement img, double maxWidth, double maxHeight) {
+    var width = img.width!;
+    var height = img.height!;
+
+    // calculate the width and height, constraining the proportions
+    if (width > height) {
+      if (width > maxWidth) {
+        height = ((height * maxWidth) / width).round();
+        width = maxWidth.toInt();
+      }
+    } else {
+      if (height > maxHeight) {
+        width = ((width * maxHeight) / height).round();
+        height = maxHeight.toInt();
+      }
+    }
+    return [width, height];
+  }
+}
+
+extension on html.File {
+  XFile toXFile({String? path}) => XFile(
+        path ?? html.Url.createObjectUrl(this),
+        name: this.name,
+        length: this.size,
+        lastModified: DateTime.fromMillisecondsSinceEpoch(
+          this.lastModified ?? DateTime.now().millisecondsSinceEpoch,
+        ),
+        mimeType: this.type,
+      );
```

Comment:
Let's not apply this extension on a dart:html class. This will cause people to some times have a `.toXFile` method on their `html.File`, and some times **not** have it, and the cause is whether they use the image_picker_for_web or not.

(This might be more appropriate for the cross_file package, but I think the best solution is still a XFile constructor for web only that takes an html.File input file.)